src/DelayId.cc

```diff
@@ -90,22 +90,13 @@ DelayId::DelayClient(ClientHttpRequest * http, HttpReply *reply)
             ch.reply = reply;
             HTTPMSGLOCK(reply);
         }
-#if FOLLOW_X_FORWARDED_FOR
-        if (Config.onoff.delay_pool_uses_indirect_client)
-            ch.src_addr = r->indirect_client_addr;
-        else
-#endif /* FOLLOW_X_FORWARDED_FOR */
-            ch.src_addr = r->client_addr;
-        ch.my_addr = r->my_addr;
-
-        if (http->getConn() != NULL)
-            ch.conn(http->getConn());
+        ch.configureClientAddr(Config.onoff.delay_pool_uses_indirect_client);
```

Comment:
The Old code used `r->client_addr` when Config.onoff.delay_pool_uses_indirect_client was off, regardless of whether Config.onoff.acl_uses_indirect_client was set. The new code checks both configuration options. AFAICT, only the old behavior was correct. There is another change request that prompted the same question: Should we refactor to call forceIndirectAddr() here instead (when needed)?